### ISSUE TITLE
Errors when automating stop_charging service

### ISSUE THREAD

**Issue by DaveGalbraith** [msp1974/homeassistant-jlrincontrol#45]
I have an iPace and I am looking to automate charging cycles, however, I am having trouble getting the stop_charging service to work. It works fine when I Call Service in Developer Tools as set up below:

![image](https://user-images.githubusercontent.com/30226844/92091850-3add1c00-edc9-11ea-91ee-095d720e2c21.png)

However, if i set it up similarly in the Automations UI like this:

![image](https://user-images.githubusercontent.com/30226844/92092142-9c9d8600-edc9-11ea-9f1b-f9f1e4d5374d.png)

I get the following errors:

2020-09-03 09:11:41 ERROR (MainThread) [homeassistant.components.automation] Jaguar Stop Charging: Error executing script. Invalid data for call_service at pos 1: value should be a string for dictionary value @ data['entity_id']
2020-09-03 09:11:41 ERROR (MainThread) [homeassistant.components.automation] While executing automation automation.jaguar_stop_charging
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 409, in async_trigger
    await self.action_script.async_run(variables, trigger_context)
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 756, in async_run
    await asyncio.shield(run.async_run())
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 173, in async_run
    await self._async_step(log_exceptions=False)
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 181, in _async_step
    await getattr(
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 370, in _async_call_service_step
    await service_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1204, in async_call
    processed_data = handler.schema(service_data)
  File "/usr/local/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 272, in __call__
    return self._compiled([], data)
  File "/usr/local/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 594, in validate_dict
    return base_validate(path, iteritems(data), out)
  File "/usr/local/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 432, in validate_mapping
    raise er.MultipleInvalid(errors)
voluptuous.error.MultipleInvalid: value should be a string for dictionary value @ data['entity_id']

**Comment by msp1974** [comment]
I'll have a look at this over the weekend

**Comment by mariwing** [comment]
If you have set up the automation exactly as your screenshot I think you have redundant information. You should not need anything in the Service data since you have selected the enity_id already in the combo box. Try to remove it and see if it works :)

**Comment by DaveGalbraith** [comment]
Yeah, that's how I did it originally, with the same result.

**Comment by msp1974** [comment]
Can you post the whole automation section from your automation.yaml file. Seen a few issues on google where setting up through gui can cause incorrect formatting in output yaml.

**Comment by DaveGalbraith** [comment]
I'm away at the mo, I'll upload when I get back at end of week, ta.

**Comment by DaveGalbraith** [comment]
As requested, here is the yaml (ignore the trigger, I just put in a dummy one):

`- id: '1599116461660'
  alias: Jaguar Stop Charging
  description: ''
  trigger:
  - event: start
    platform: homeassistant
  condition: []
  action:
  - data: {}
    entity_id: sensor.jag_info
    service: jlrincontrol.stop_charging
  mode: single`

**Comment by msp1974** [comment]
Dave, this is hard to see if there is a format error. Can you post as code segment.

**Comment by DaveGalbraith** [comment]
Hmm, I'm not sure what you want me to do here, I did use Insert Code and noticed that the indentation was odd, but there doesn't seem to be any other option.

**Comment by msp1974** [comment]
Post a screen shot

**Comment by DaveGalbraith** [comment]
![Screenshot_20200918-073551_Home Assistant](https://user-images.githubusercontent.com/30226844/93564536-c33eed80-f981-11ea-9235-8473f9e15d22.jpg)

**Comment by msp1974** [comment]
So this looks fine. I cant replicate this issue but i also dont have an EV to be able to properly test.  Can you try a couple of things and give me some more info.

Can you try and automate a different servicewith same entity and also different entity. Can you also try and use a different trigger that HA start to see if it is caused by the component not being fully loaded when HA start triggers.

Can you also tell me version of HA you are running.

**Comment by DaveGalbraith** [comment]
Ok, so I tried :

Changing service to jlrcontrol.start_charging 
Changing entity to sensor.jag_ev_battery

and got the same result and same errors as I originally had.

HA version is 0.113.0

**Comment by DaveGalbraith** [comment]
Oh and my original automation did not use the HA Start trigger, it was a time based trigger (I created the latest automation as a trial just to rule out the other things I had in the original)

**Comment by DaveGalbraith** [comment]
And just in case it helps, I've just upgraded to 0.115.1 and I still get the same result.

**Comment by msp1974** [comment]
Ok.  So I have managed to recreate this and it seems it does the same this for all services.  It may take me a few days to debug and work out a fix but it is definitely some error in the component.  Just not sure what yet! ;-)

**Comment by DaveGalbraith** [comment]
Cool no worries, I'm glad you managed to replicate it, that's half the battle.

**Comment by msp1974** [comment]
Actually quicker (and easier than I thought).  I think this may be some issue with HA as when tried some of my other integrations, I got the same error - which is actually a bit misleading.  It is not sending and array instead of a string but sending nothing for entity id.

Are you seeing the same? It maybe that something has changed and it needs updating, however, I have a work around that will get this working for you, which I found from looking at automations that I have created by hand.

Basically, this issue is the format of the yaml created by the automation UI.  In your action section, change it to...

![image](https://user-images.githubusercontent.com/11211952/93708351-4e101b80-fb2d-11ea-9ddf-1bd0261bcfb8.png)

Let me know how you get on.

**Comment by DaveGalbraith** [comment]
Well I no longer get the error, but then, neither does it do anything (car just carries on charging).

**Comment by msp1974** [comment]
Does it show in the logs (need to set integration to debug level in configuration.yaml) that it is calling the service?

![image](https://user-images.githubusercontent.com/11211952/93709193-240e2780-fb34-11ea-987a-0979abd48fc4.png)

**Comment by DaveGalbraith** [comment]
Assuming I should be looking in the Homeassistant Logs, then no, I don't see it being called.

**Comment by DaveGalbraith** [comment]
Soz, there was a typo in my logger entry, it does call the service but it fails:

2020-09-20 14:28:20 INFO (MainThread) [custom_components.jlrincontrol.services] Service charging_stop called on vehicle Jag. Awaiting feedback on success.
2020-09-20 14:28:21 INFO (MainThread) [custom_components.jlrincontrol.services] InControl service call (charging_stop) to vehicle Jag failed due to None. 
Full return is {'status': 'Running', 'statusTimestamp': '2020-09-20T13:28:21+0000', 'startTime': '2020-09-20T13:28:20+0000', 'serviceType': 'CP', 'failureDescription': '', 'customerServiceId': 'redacted', 'vehicleId': 'redacted', 'active': True, 'initiator': 'USER', 'eventTrigger': None, 'serviceCommand': None, 'serviceParameters': [{'key': 'CHARGE_NOW_SETTING', 'value': 'FORCE_OFF'}]}

**Comment by msp1974** [comment]
Ok well some progress i guess. Just to confirm, this works when you run via developer tab and does stop the car charging?

This looks like it has called the service and sent to the car but it is not expecting a status of 'running' back from the jlr servers. It loops when status is 'started' and waits for a status other than that. If it gets a 'success' status then thats good, anything else is a fail.  This is not the final status and im guessing it needs to keep looping if status is running to wait for the final status message.

Did it actually stop charging at all?

**Comment by DaveGalbraith** [comment]
You ask an interesting question. It previously worked with the developer tab but I haven't tried that again this time. However, I did try stopping it with the (generally useless) Jaguar app and (the much better) WattCat app. Neither worked. It never stopped charging but I'm wondering if the API calls are generally flaky. I'll experiment a bit more during the week and let you know.

**Comment by DaveGalbraith** [comment]
As suspected, I am not able to stop charging using any of developer tab, HA automation or  either of the apps. I suspect you have fixed the original issue I had using the stop_charging service and now we are up against flaky JLR software.
